lib/components/overlays/quick_settings/quick_settings_overlay.dart

```diff
@@ -117,6 +124,49 @@ class _QuickSettingsOverlayState extends State<QuickSettingsOverlay>
   }
 }
 
+PageRouteBuilder customRouteTransition(Widget screen) {
+  return PageRouteBuilder(
+    transitionDuration: Constants.animationDuration,
+    reverseTransitionDuration: Constants.animationDuration,
+    pageBuilder: (context, animation, secondaryAnimation) => screen,
+    transitionsBuilder: (context, animation, secondaryAnimation, child) {
+      final fadeAnimationIn = Tween(
+        begin: animation.value,
+        end: animation.value,
+      ).animate(animation);
```

Comment:
final fadeAnimationIn = animation;

What you have here will break hard

Reply:
This actually works because tweens are mutable and update when their value changes. It's not very intuitive though...

I kinda liked the way it looked before because it explicitly shows what values we're looping through and allows for better customisation later down the line.

Though if this is really important to you, idm changing it to final fadeAnimationIn = animation;

Reply:
This is barely understandable when reading, i had no idea it would work before you telling me even tho I knew about tweens being mutable. The straight up assignment makes it clear that the fadeIn animation is just the normal linear animation, not more, not less

Reply:
Ok, done